.github/workflows/py-checks.yml

```diff
@@ -19,47 +24,64 @@ jobs:
       matrix:
         python-version: ["3.10"]
     steps:
-    - uses: actions/checkout@v3
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v3
-      with:
-        python-version: ${{ matrix.python-version }}
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install -r ./requirements.dev.txt
-    - name: Check with black
-      run: |
-        VERNUM=${{ matrix.python-version }}
-        TARGETV="py${VERNUM/./}"
-        python -m black -t $TARGETV --diff --color --check ./
-    - name: Check imports with isort (black)
-      run: |
-        python -m isort --profile=black --check --diff --color ./
-    - name: Check with flake8
-      run: |
-        python -m flake8 ./
-    - name: Check with Pylint
-      run: |
-        pylint --recursive=y .
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Cache pip dependencies for lint/format
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/pip
+          key: lint-${{ runner.os }}-pip-${{ matrix.python-version }}-${{ hashFiles('requirements.dev.txt') }}-${{ github.run_id }}
+          restore-keys: |
+            lint-${{ runner.os }}-pip-${{ matrix.python-version }}-${{ hashFiles('requirements.dev.txt') }}-
+            lint-${{ runner.os }}-pip-${{ matrix.python-version }}-
```

Comment:
I'm not sure we need the less strict key as a fall-back here.   Wouldn't this effectively make non-existing caches just go back and use the old existing cache?